keras_cv/metrics/coco/base.py

```diff
@@ -225,6 +225,21 @@ def _match_boxes(self, y_true, y_pred, threshold, ious):
             gt_matches = gt_matches.write(match_index, detection_idx)
         return pred_matches.stack()
 
+    @property
+    def mergeable_weights(self):
+        return [self.true_positives, self.false_positives, self.ground_truth_boxes]
+
+    def merge_state(self, metrics):
+        assign_add_ops = []
+        for metric in metrics:
+            if len(self.mergeable_weights) != len(metric.mergeable_weights):
+                raise ValueError(f"Metric {metric} is not compatible with {self}")
+            for weight, weight_to_add in zip(
+                self.mergeable_weights, metric.mergeable_weights
+            ):
+                assign_add_ops.append(weight.assign_add(weight_to_add))
```

Comment:
I think we should check the shape of the weights, so that we don't accidentally broadcast the weights and produce wrong results.